src/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,23 @@
+import { useState } from "react";
+import { LocalStorageKey } from "../enum";
+
+const useLocalStorage = () => {
+    //TODO: Implement its useEffect when context is ready
+    const [events] = useState<MainEvents>(() => getFromStorage<MainEvents>(LocalStorageKey.Events) ?? []);
```

Comment:
mention the reason of why these are placed in useState, if the above comment is the reason its not clear, update it